default/scripting/species/common/xenophobic.macros

```diff
@@ -6,46 +6,46 @@ XENOPHOBIC_SELF
             activation = Turn high = 0
             effects = GiveEmpireTech name = "CON_CONC_CAMP"
 
-        // Give malus to industry, research and happiness for other species nearby
+        // Stability malus for other species nearby
         EffectsGroup
             scope = Source
             activation = And [
                 Planet
                 Number low = 1 condition = And [
                     Planet
-                    OwnedBy empire = RootCandidate.Owner
-                    Not Or [ 
-                        Species name = RootCandidate.Species
-                        Species name = "SP_EXOBOT"
-                    ]
+                    Not Species name = RootCandidate.Species
+                    Not Species name = "SP_EXOBOT"
                     Not Population high = 0
                     Not Contains Building name = "BLD_CONC_CAMP"
-                    WithinStarlaneJumps jumps = 5 condition = Source
+                    Or [
+                        Not OwnedBy empire = RootCandidate.Owner
+                        Not EmpireHasAdoptedPolicy empire = Source.Owner name = "PLC_RACIAL_PURITY"
+                    ]
+                    WithinStarlaneJumps jumps = (NamedReal name = "XENOPHOBIC_MAX_JUMPS" value = 4) condition = Source
```

Comment:
Sounds like a good case for a NamedInteger

Reply:
In the forum you wrote that only visible planets should trigger it, but I do not see a condition for it. Doesn't this sometimes produce unknown values?